viewer-ng/src/test/java/nl/tailormap/viewer_ng/HSQLDBTestProfileJPAConfiguration.java

```diff
@@ -0,0 +1,56 @@
+package nl.tailormap.viewer_ng;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.env.Environment;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+import org.springframework.transaction.PlatformTransactionManager;
+
+import javax.persistence.EntityManagerFactory;
+import javax.sql.DataSource;
+
+@Configuration
+@EntityScan(
+        basePackages = {"nl.tailormap.viewer.config", "nl.tailormap.viewer_ng.repository"}
+)
+public class HSQLDBTestProfileJPAConfiguration {
+
+    @Autowired
+    private Environment env;
+
+    @Bean
+    @Profile("test")
+    public LocalContainerEntityManagerFactoryBean entityManagerFactory() {
+        LocalContainerEntityManagerFactoryBean em = new LocalContainerEntityManagerFactoryBean();
+        em.setPersistenceUnitName("viewer-config-hsqldb");
+        em.setDataSource(dataSource());
+
+        return em;
+    }
+
+    @Bean
+    @Profile("test")
+    public DataSource dataSource() {
```

Comment:
This is not necessary when having a separate test propertiesfile (which you have) and you add a annotation to the test-class: 
@TestPropertySource(
        locations = "classpath:application-integrationtest.properties").

When using the default propertynames + hsqldb dependency in you pom file, springboot will automatically create the datasource.

Reply:
yeah need to figure that out, needed to upgrade to JUnit 5 first (#2827 )

Reply:
it turns out I need to override some things that are specified in the persistence.xml; I have something working now that (for each test):
1. creates a database with a unique name
2. loads the test data
so intergations tests can run multithreaded